src/libraries/Microsoft.Extensions.Logging.Console/src/ContainerUtils.cs

```diff
@@ -0,0 +1,53 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Extensions.Logging.Console
+{
+    internal static class ContainerUtils
+    {
+        private const string RunningInContainerVariableName = "DOTNET_RUNNING_IN_CONTAINER";
+        private const string DeprecatedRunningInContainerVariableName = "DOTNET_RUNNING_IN_CONTAINERS";
+
+        public static bool IsContainer { get; } = IsProcessRunningInContainer();
+
+        public static bool IsProcessRunningInContainer()
+        {
+            // Official .NET Core images (Windows and Linux) set this. So trust it if it's there.
+            // We check both DOTNET_RUNNING_IN_CONTAINER (the current name) and DOTNET_RUNNING_IN_CONTAINERS (a deprecated name used in some images).
```

Comment:
> a deprecated name used in some images

Will this code end up being used in such old images?

Reply:
@shirhatti

Reply:
I think it is **extremely unlikely** this code will ever end up in such an old image